client/landing/stepper/README.md

```diff
@@ -1,106 +1,387 @@
+	const hasAnySites = userHasAnySites( calypsoReduxStore.getState() );
 
-export const exampleFlow: Flow = {
-	initialize() {
-		return [ STEPS.DOMAINS, STEPS.DESIGN ];
-	},
-	useStepNavigation( currentStep, navigate ) {
-		const goBack = () => {
-			if ( currentStep === 'domain' ) {
-				navigate( 'design' );
-			} else {
-				navigate( 'domain' );
+	if ( includeDomainsStep ) {
+		return [ STEPS.UNIFIED_DOMAINS, STEPS.UNIFIED_PLANS, STEPS.PROCESSING ] as const;
+	}
+
+	if ( hasAnySites ) {
+		return [ STEPS.PICK_SITE, STEPS.UNIFIED_DOMAINS, STEPS.UNIFIED_PLANS, STEPS.PROCESSING ] as const;
+	}
+
+	// We need `as const` to promise TS that these steps won't change later.
+	return [ STEPS.UNIFIED_PLANS, STEPS.PROCESSING ] as const;
+}
+
+export const exampleFlow: FlowV2< typeof initialize > = {
+	/**
+	 * The name of the flow is what appears in the pathname. It must be unique.
+	 * This flow will be under /setup/my-flow.
+	 */
+	name: 'my-flow',
+
+	/**
+	 * This flag must be `true` for signup flows (generally where a new site may be created), and should be `false` for other flows.
+	 * It controls whether we'll trigger a `calypso_signup_start` Tracks event when the flow starts.
+	 */
+	isSignupFlow: true,
+	initialize,
+	/**
+	 * This hook is the control unit of your flow. It is where:
+	 * 1. You handle and use the information submitted by the steps.
+	 * 2. You decide to navigate to which steps when.
+	 * 3. You manage the state of the flow.
+	 */
+	useStepNavigation( currentStepSlug, navigate ) {
+		// This hook should be enough to manage and persist all the state your flows needs.
+		const { get, set } = useFlowState();
+
+		// This function handles the submission event of every step in the flow.
+		const submit: SubmitHandler< typeof initialize > = async ( submittedStep ) => {
+			const { slug, providedDependencies } = submittedStep;
+			switch ( slug ) {
+				case 'domains': {
+					// Here we have the data submitted by the domains step.
+					// `providedDependencies` will be fully-typed and should contain the exact types of the domain step submission.
+					const domainItem = providedDependencies.domainItem;
+					// By calling this, we're updating the state of the flow to save the picked domain.
+					// This will be precisely persisted for the whole session, but not longer.
+					set( 'domains', providedDependencies );
+					// Because the flow knows the steps it contains, `navigate` will only only allow `'plans' | 'domains' | 'processing'` are the first argument.
+					navigate( 'plans' );
+					break;
+				}
+				case 'plans': {
+					// Here we have the data submitted by the plans step.
+					set( 'plans', providedDependencies );
+
+					// setPendingAction allows you to enqueue any promise.
+					setPendingAction( () => createSite() );
+					break;
+				}
+				case 'processing': {
+					// The processing step will pick up the pending action we set above, run it, and await it.
+					// It will show a progress bar during that time.
```

Comment:
Does this mean the `processing` step is a special one? Maybe we should clarify it!

Reply:
Yes! That's why I, and Gabriel, hate it, and looking to get rid of it ( paYKcK-62X-p2 ) .